packages/core/test/unit/sql/delete.test.js

```diff
@@ -18,60 +18,30 @@ describe(Support.getTestDialectTeaser('SQL'), () => {
       schema: 'public',
     });
 
-    describe('truncate #4306', () => {
+    describe('delete with model', () => {
       const options = {
-        table: User.getTableName(),
-        where: {},
-        truncate: true,
-        cascade: true,
-        limit: 10,
-        type: QueryTypes.BULKDELETE,
-      };
-
-      it(util.inspect(options, { depth: 2 }), () => {
-        return expectsql(
-          sql.truncateTableQuery(
-            options.table,
-            options,
-          ), {
-            ibmi: 'TRUNCATE TABLE "public"."test_users" IMMEDIATE',
-            postgres: 'TRUNCATE "test_users" CASCADE',
-            mssql: 'TRUNCATE TABLE [public].[test_users]',
-            mariadb: 'TRUNCATE `public`.`test_users`',
-            mysql: 'TRUNCATE `public`.`test_users`',
-            db2: 'TRUNCATE TABLE "public"."test_users" IMMEDIATE',
-            sqlite: 'DELETE FROM `public.test_users`',
-            snowflake: 'TRUNCATE "public"."test_users"',
-          },
-        );
-      });
-    });
-
-    describe('truncate with cascade and restartIdentity', () => {
-      const options = {
-        table: User.getTableName(),
-        where: {},
+        table: User,
+        where: { name: 'foo' },
         truncate: true,
-        cascade: true,
-        restartIdentity: true,
-        limit: 10,
         type: QueryTypes.BULKDELETE,
       };
 
       it(util.inspect(options, { depth: 2 }), () => {
         return expectsql(
-          sql.truncateTableQuery(
+          sql.deleteQuery(
```

Comment:
This test does not seem to be testing anything that isn't already tested in `unit/query-generator/delete-query.test.ts` and could be deleted

Reply:
It is testing a few more `deleteQuery` scenarios compared to `unit/query-generator/delete-query.test.ts`.
Probably best to wait till `deleteQuery` is migrated to typescript unless you want me to mirgrate it in this PR.